samwise.js

```diff
@@ -1,56 +1,55 @@
 /**
```

Comment:
SCRIPT-This function might benefit from using async/await instead of callbacks for better readability.